### PR TITLE
Apply IPYTHON conf.prompt

### PR DESCRIPTION
**Pull request by gpotter2**
This PR fixes:
- conf.prompt not beeing updated https://github.com/secdev/scapy/pull/665#issuecomment-331145295
- save_session/load_session not using ipython user_ns var.
- python 3 session tests fixed

first bug fixes:
```
>>> conf.prompt = 'toto >'
toto >
toto >VERSION
'2.3.3.dev695'
toto >
```

### CODE REVIEW

**Comment by guedou** [comment]
Could you provide an example ?

File: scapy/error.py [hunk 1]
```diff
@@ -64,13 +64,13 @@ def emit(self, record):
...
 
-def warning(x, *args, **kargs):
+def warning(x, onlyOnce=False, *args, **kargs):
     """
     Prints a warning during runtime.
 
     onlyOnce - if True, the warning will never be printed again.
     """ 
-    if kargs.pop("onlyOnce", False):
+    if kargs.pop("onlyOnce", False) or onlyOnce:
```
Comment (ThomasFaivre) [inline scapy/error.py]: What is the purpose of this change? Also, if you explicitly put onlyOnce in the arguments, it cannot be in kwargs.

File: scapy/themes.py [hunk 2]
```diff
@@ -288,11 +288,18 @@ class HTMLTheme2(HTMLTheme):
...
     from IPython.terminal.prompts import Prompts, Token
-    shell.highlighting_style_overrides = {
+    from scapy.config import conf
+    class ClassicPrompt(Prompts):
+        def in_prompt_tokens(self, cli=None):
+           return [(Token.Prompt, conf.prompt),]
+        def out_prompt_tokens(self):
+           return [(Token.OutPrompt, ''),]
+    shell.prompts_class=ClassicPrompt # Set classic prompt style
+    shell.highlighting_style_overrides = { # Register and apply scapy color style
```
Comment (ThomasFaivre) [inline scapy/themes.py]: It could be interesting to allow the definition of a Prompt class for conf.prompt:

```
    if isinstance(conf.prompt, basestring):
        class ClassicPrompt(Prompts):
            def in_prompt_tokens(self, cli=None):
                return [(Token.Prompt, conf.prompt),]
            def out_prompt_tokens(self):
                return [(Token.OutPrompt, ''),]
        shell.prompts_class = ClassicPrompt # Set classic prompt style
    else:
        shell.prompts_class = conf.prompt
```

File: scapy/config.py [hunk 3]
```diff
@@ -329,11 +329,11 @@ def isCryptographyAdvanced():
 def _prompt_changer(attr, val):
     """Change the current prompt theme"""
     try:
-        sys.ps1 = val.prompt(conf.prompt)
+        sys.ps1 = conf.color_theme.prompt(conf.prompt)
```
Comment (gpotter2) [inline scapy/config.py]: sys.ps1 is used when IPython is not available :-)

File: scapy/themes.py (hunk 2 above)
Comment (gpotter2) [reply scapy/themes.py]: Sounds good

File: scapy/error.py (hunk 1 above)
Comment (gpotter2) [reply scapy/error.py]: Right, I'll do that 
That's currently a bug while scapy calls ˋwarning("msg", True)` since **kwargs were added to the warning function. But now kargs.pop is useless

**Comment by gpotter2** [comment]
Ready for review.

**Comment by ThomasFaivre** [comment]
Good for me!

**Comment by gpotter2** [comment]
@p-l- This also fixes the python 3 failing tests about sessions. Please ignore coverage
